Controller/Adminhtml/Landingpage/Delete.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Controller\Adminhtml\Landingpage;
+
+use Magento\Framework\Controller\ResultFactory;
+
+class Delete extends AbstractAction
+{
+    /** @return \Magento\Framework\View\Result\Page */
+    public function execute()
+    {
+        /** @var \Magento\Backend\Model\View\Result\Redirect $resultRedirect */
+        $resultRedirect = $this->resultFactory->create(ResultFactory::TYPE_REDIRECT);
+
+        $landingPageId = $this->getRequest()->getParam('id');
+        if ($landingPageId) {
+            try {
+                /** @var \Algolia\AlgoliaSearch\Model\LandingPage $landingPage */
+                $landingPage = $this->landingPageFactory->create();
+                $landingPage->getResource()->load($landingPage, $landingPageId);
+                $landingPage->getResource()->delete($landingPage);
```

Comment:
URL is deleted automatically?

Reply:
Yes, I added a _afterDelete() method on the resource model : https://github.com/algolia/algoliasearch-magento-2/blob/feat-landing-page-builder/Model/ResourceModel/LandingPage.php#L79